src/components/TimelineComponent.tsx

```diff
@@ -415,29 +446,30 @@ export default class TimelineComponent
    * said timeline
    */
   drawTimeline() {
-    const zoom = d3.zoom()
+    this.zoom = d3.zoom()
         .scaleExtent([1, 20]) // zoom range
```

Comment:
drawTimeline() is a big method, could be split up? Definitely more inline commenting would be useful